content/blog/CustomWebApiComponent/index.md

```diff
@@ -0,0 +1,384 @@
+
+```sh
+$ tree
+.
+├── pom.xml
+├── src
+│   ├── main
+│   │   ├── docs
+│   │   │   └── chuck-component.adoc
+│   │   └── java
+│   │       └── org
+│   │           └── apache
+│   │               └── camel
+│   │                   └── component
+│   │                       └── chuck
+│   └── test
+│       ├── java
+│       │   └── org
+│       │       └── apache
+│       │           └── camel
+│       │               └── component
+│       │                   └── chuck
+│       └── resources
+│           └── log4j2.properties
+```
+
+## Component creation
+
+After the initial project setup, we are ready to add the component main building blocks.
+
+When the Camel Context starts, it creates the Component object, that creates the Endpoint object, that in turns creates the Producer and the Consumer objects as needed. The Camel API includes convenient default implementation for all of them, that are also easy to extend.
+
+Our component class extends the DefaultComponent class and we also add a configuration class to hold component's query parameters that will allow some level of customization.
+
+```java
+@Component("chuck")
+public class ChuckComponent extends DefaultComponent {
+    @Override
+    protected Endpoint createEndpoint(final String uri, String remaining, final Map<String, Object> parameters) throws Exception {
+        final ChuckConfiguration configuration = new ChuckConfiguration();
+        //…
+    }
+}
+```
+
+Then, we need to extend the DefaultEndpoint and the DefaultAsyncProducer to be able to support asynchronous HTTP requests. For the vast majority of web APIs we don't need to create a Consumer, unless you have a polling endpoint. In that case, you would need to extend ScheduledPollEndpoint and ScheduledPollConsumer.
+
+```java
+@UriEndpoint(firstVersion = "3.0.0", scheme = "chuck", title = "Chuck", syntax = "chuck:type", label = "chuck", producerOnly = true)
+public class ChuckEndpoint extends DefaultEndpoint {
+    @UriParam
+    private ChuckConfiguration configuration;
+    //…
+    @Override
+    protected void doStart() throws Exception {
+        //…
+    }
+    @Override
+    protected void doStop() throws Exception {
+        //…
+    }
+    @Override
+    public Producer createProducer() throws Exception {
+        return new ChuckProducer(this);
+    }
+    @Override
+    public Consumer createConsumer(Processor processor) throws Exception {
+        throw new UnsupportedOperationException("No support for consumers");
+    }
+}
+
+public class ChuckProducer extends DefaultAsyncProducer {
+    @Override
+    public boolean process(Exchange exchange, AsyncCallback callback) {
+        //…
+        return false;
+    }
+}
+```
+
+In the process method of the Producer we return false to signal that the processing will be executed asynchronously. Look at the repository for the full source code, but these are the only classes needed to start building a new component. 
```

Comment:
Here it might be a problem as well. It wasn't reported but just to make sure.

```suggestion
In the process method of the Producer we return false to signal that the processing will be executed asynchronously. Look at the repository for the full source code, but these are the only classes needed to start building a new component.
```